papermill_origami/engine.py

```diff
@@ -106,6 +106,11 @@ async def execute(self, **kwargs):
 
         dagster_logger = kwargs["logger"]
 
+        # we can put data in here for the calling function to access after execution
+        engine_out_data = kwargs.get("engine_out_data", {})
```

Comment:
Instead of mutating an external dict, I'd say we should instead add any arbitrary metadata into the `self.nb` notebook instance, and access that from the return value of `pm.execute_notebook`.

So, instead of 

```python
        engine_out_data["executed_notebook_url"] = parameterized_url
        engine_out_data["parameterized_notebook_id"] = self.file.id
```

We'd do something like,

```python
self.nb.metadata["executed_notebook_url"] = parameterized_url
self.nb.metadata["parameterized_notebook_id"] = self.file.id
```

Or, you could nest them all under `engine_out_data` in the metadata, up to you! :)

Reply:
oh cool! that's a much better pattern. didn't realize i could add metadata to the notebook object, will definitely do that